pkg/crosscluster/physical/stream_ingestion_processor.go

```diff
@@ -1064,6 +1075,39 @@ func (r *rangeKeyBatcher) flush(ctx context.Context, toFlush mvccRangeKeyValues)
 	return nil
 }
 
+func (r *rangeKeyBatcher) recomputeStats(ctx context.Context, rangekeySpans roachpb.Spans) {
+	rangeStartKeysFound := make(map[string]struct{})
+	rangeStartKeys := make([]roachpb.RKey, 0)
+
+	for i := range rangekeySpans {
+		iter, err := r.rangeDescIterFactory.NewLazyIterator(ctx, rangekeySpans[i], 100)
```

Comment:
What do you think of splitting `rangeStartKeys` into a function and then writing a unit test for it? That lets us test the complicated part of this function and it makes the code a little cleaner.

Reply:
Nice idea! Done.